main.py

```diff
@@ -81,12 +81,33 @@ def swap_track(track_id):
     return [11, 14, 9, 16, 15, 10, 13, 12][track_id - 9]
 
 
+def tail_call_gen(func: typing.Callable[[...], typing.Generator]):
+    """
```

Comment:
I don't know if you've come across this syntax before but it's a docstring, which is essentially just a comment that's attached to a callable (i.e a class, a function or a method) to document what it does, this helps to differentiate versus comments about why a line of code is doing something a particular way